coderd/userauth_test.go

```diff
@@ -1654,6 +1655,48 @@ func TestOIDCSkipIssuer(t *testing.T) {
 	require.Equal(t, found.LoginType, codersdk.LoginTypeOIDC)
 }
 
+func TestUserForgotPassword(t *testing.T) {
```

Comment:
Knowing only that functionality works is insufficient; how it _fails_ is just as important.
In fact, this is not even testing that the password is changed, only that a notification is sent to do so and that it doesn't error out.

For this functionality to be appropriately tested, it would need to simulate a failed login, execute the change, and then have the login succeed.

Over and above that, there are numerous ways in which this can break, and we have to cover as much of the API with tests as is practical.

---

As a follow-up, if you're interested, consider adding a CI check which prints the coverage for the new code when submitting a PR would be very cool. It would've shown you here that your testing was incomplete.